### PR TITLE
feat: Support non-default cluster domain

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- bump to op-rs 0.79
- make cluster domain configurable

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Comment by sbernauer** [comment]
Running tests
kind https://testing.stackable.tech/job/zookeeper-operator-it-custom/2
AKS https://testing.stackable.tech/job/zookeeper-operator-it-custom/3
OpenShift https://testing.stackable.tech/job/zookeeper-operator-it-custom/4
GKE https://testing.stackable.tech/job/zookeeper-operator-it-custom/5
EKS https://testing.stackable.tech/job/zookeeper-operator-it-custom/6
IONOS https://testing.stackable.tech/job/zookeeper-operator-it-custom/7

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -483,10 +484,14 @@ impl ZookeeperCluster {
 
     /// The fully-qualified domain name of the role-level load-balanced Kubernetes `Service`
     pub fn server_role_service_fqdn(&self) -> Option<String> {
+        let cluster_domain = KUBERNETES_CLUSTER_DOMAIN
+            .get()
+            .expect("Could not resolve the Kubernetes cluster domain!");
```
Comment (sbernauer) [inline rust/crd/src/lib.rs:489]: I would prefer the longer message
```suggestion
            .expect("KUBERNETES_CLUSTER_DOMAIN must first be set by calling initialize_operator");
```

File: rust/operator-binary/src/zk_controller.rs [hunk 2]
```diff
@@ -801,9 +812,13 @@ fn build_server_rolegroup_statefulset(
             ..EnvVar::default()
         }])
         .add_volume_mount("data", STACKABLE_DATA_DIR)
+        .unwrap()
```
Comment (sbernauer) [inline rust/operator-binary/src/zk_controller.rs:815]: Can we please use error handling everywhere?

File: rust/operator-binary/src/zk_controller.rs [hunk 3]
```diff
@@ -860,10 +875,15 @@ fn build_server_rolegroup_statefulset(
         .add_container_port("zk-election", 3888)
         .add_container_port("metrics", 9505)
         .add_volume_mount("data", STACKABLE_DATA_DIR)
+        .unwrap()
```
Comment (sbernauer) [inline rust/operator-binary/src/zk_controller.rs:878]: Can we please use error handling everywhere?

File: rust/operator-binary/src/zk_controller.rs [hunk 4]
```diff
@@ -921,38 +944,45 @@ fn build_server_rolegroup_statefulset(
...
+                "log",
+                logging.containers.get(&Container::Vector),
+                ResourceRequirementsBuilder::new()
+                    .with_cpu_request("250m")
+                    .with_cpu_limit("500m")
+                    .with_memory_request("128Mi")
+                    .with_memory_limit("128Mi")
+                    .build(),
+            )
+            .unwrap(),
```
Comment (sbernauer) [inline rust/operator-binary/src/zk_controller.rs:984]: Can we please use error handling everywhere?

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: rust/operator-binary/src/zk_controller.rs (hunk 2 above)
Comment (maltesander) [reply rust/operator-binary/src/zk_controller.rs]: Thanks, done https://github.com/stackabletech/zookeeper-operator/pull/870/commits/59e2ea90cc3841c42a4e360079b2e0dfc91306c7

File: rust/operator-binary/src/zk_controller.rs (hunk 3 above)
Comment (maltesander) [reply rust/operator-binary/src/zk_controller.rs]: Thanks, done https://github.com/stackabletech/zookeeper-operator/pull/870/commits/59e2ea90cc3841c42a4e360079b2e0dfc91306c7

File: rust/operator-binary/src/zk_controller.rs (hunk 4 above)
Comment (maltesander) [reply rust/operator-binary/src/zk_controller.rs]: Thanks, done https://github.com/stackabletech/zookeeper-operator/pull/870/commits/59e2ea90cc3841c42a4e360079b2e0dfc91306c7

File: rust/operator-binary/src/zk_controller.rs [hunk 5]
```diff
@@ -323,7 +346,7 @@ pub async fn reconcile_zk(
         &zk.object_ref(&()),
         ClusterResourceApplyStrategy::from(&zk.spec.cluster_operation),
     )
-    .unwrap();
+    .context(CreateClusterResourcesSnafu)?;
```
Comment (sbernauer) [inline rust/operator-binary/src/zk_controller.rs:349]: Thx!

**Review by sbernauer — APPROVED** [review]

**Pull request merged**